doc/src/manual/code-loading.md

```diff
@@ -0,0 +1,321 @@
+
+- **paths:** `uuid::UUID` × `name::Symbol` ⟶ `path::String`
+
+   The `paths` map assigns to each package UUID-name pair, the location of the entry-point source file of that package. After the identity of `X` in `import X` has been resolved to a UUID via `roots` or `graph` (depending on whether it is loaded from the main project or an dependency), Julia determines what file to load to acquire `X` by looking up `paths[uuid,:X]` in the environment. Including this file should create a module named `X`. After the first time this package is loaded, any import resolving to the same `uuid` will simply create a new binding to the same already-loaded package module.
+
+Each kind of environment defines these three maps differently, as detailed in the following sections.
+
+### Project environments
+
+A project environment is determined by a directory containing a project file, `Project.toml`, and optionally a manifest file, `Manifest.toml`. These files can also be named `JuliaProject.toml` and `JuliaManifest.toml`, in which case `Project.toml` and `Manifest.toml` are ignored; this allows for coexistence with other tools that might consider files named `Project.toml` and `Manifest.toml` significant. For pure Julia projects, however, the names `Project.toml` and `Manifest.toml` should be preferred. The `roots`, `graph` and `paths` maps of a project environment are defined as follows.
+
+**The roots map** of the environment is determined by the contents of the project file, specifically, its top-level `name` and `uuid` entries and its `[deps]` section (all optional). Consider the following example project file for the hypothetical application, `App`, as described above:
+
+```toml
+name = "App"
+uuid = "8f986787-14fe-4607-ba5d-fbff2944afa9"
+
+[deps]
+Priv = "ba13f791-ae1d-465a-978b-69c3ad90f72b"
+Pub  = "c07ecb7d-0dc9-4db7-8803-fadaaeaf08e1"
+```
+
+This project file implies the following `roots` map, if it were materialized as a Julia dictionary:
+
+```julia
+roots = Dict(
+    :App  => UUID("8f986787-14fe-4607-ba5d-fbff2944afa9"),
+    :Priv => UUID("ba13f791-ae1d-465a-978b-69c3ad90f72b"),
+    :Pub  => UUID("c07ecb7d-0dc9-4db7-8803-fadaaeaf08e1"),
+)
+```
+
+Note that for efficiency reasons, `roots` maps are not actually materialized as dictionaries when loading code and are instead queried through internal APIs. Given this `roots` map, in the code a `App` the statement `import Priv` will cause Julia to look up `roots[:Priv]`, which yields `ba13f791-ae1d-465a-978b-69c3ad90f72b`, the UUID of the `Priv` package that is to be loaded in that context. This UUID identifies which `Priv` package to load and use when the main application evaluates `import Priv`.
```

Comment:
> Note that for efficiency reasons ...

Maybe this sentence can be removed? Describing how things are implemented internally feels a bit out of place here?